apps/web/src/pages/TrainingOpportunities/components/TrainingOpportunityForm.tsx

```diff
@@ -0,0 +1,163 @@
+        nullSelection={intl.formatMessage({
+          defaultMessage: "Select a language",
+          id: "uup5F2",
+          description:
+            "Placeholder displayed on the user form preferred communication language field.",
+        })}
+        rules={{
+          required: intl.formatMessage(errorMessages.required),
+        }}
+        options={localizedEnumToOptions(courseLanguages, intl)}
+      />
+      <Select
+        id="courseFormat"
+        name="courseFormat"
+        label={intl.formatMessage(formLabels.format)}
+        nullSelection={intl.formatMessage({
+          defaultMessage: "Select a format",
+          id: "m3c4o8",
+          description: "Placeholder displayed on the select input for a format",
+        })}
+        rules={{
+          required: intl.formatMessage(errorMessages.required),
+        }}
+        options={localizedEnumToOptions(courseFormats, intl)}
+      />
+      <DateInput
+        id="applicationDeadline"
+        legend={intl.formatMessage(formLabels.applicationDeadline)}
+        name="applicationDeadline"
+        show={[DATE_SEGMENT.Day, DATE_SEGMENT.Month, DATE_SEGMENT.Year]}
+        rules={{
+          required: intl.formatMessage(errorMessages.required),
+        }}
+      />
+      <div data-h2-display="base(none) p-tablet(inherit)">
+        {/* intentionally left blank */}
+      </div>
+      <DateInput
+        id="trainingStartDate"
+        legend={intl.formatMessage(formLabels.trainingStartDate)}
+        name="trainingStartDate"
+        show={[DATE_SEGMENT.Day, DATE_SEGMENT.Month, DATE_SEGMENT.Year]}
+        rules={{
+          required: intl.formatMessage(errorMessages.required),
+        }}
+      />
+      <DateInput
+        id="trainingEndDate"
+        legend={intl.formatMessage(formLabels.trainingEndDate)}
+        name="trainingEndDate"
+        show={[DATE_SEGMENT.Day, DATE_SEGMENT.Month, DATE_SEGMENT.Year]}
+      />
+      <RichTextInput
+        id="descriptionEn"
+        label={intl.formatMessage(formLabels.descriptionEn)}
+        name="descriptionEn"
+        rules={{
+          required: intl.formatMessage(errorMessages.required),
+        }}
+      />
+      <RichTextInput
+        id="descriptionFr"
+        label={intl.formatMessage(formLabels.descriptionFr)}
+        name="descriptionFr"
+        rules={{
+          required: intl.formatMessage(errorMessages.required),
+        }}
+      />
+      <Input
+        id="applicationUrlEn"
+        name="applicationUrlEn"
+        label={intl.formatMessage(formLabels.applicationUrlEn)}
+        type="text"
+        rules={{
+          required: intl.formatMessage(errorMessages.required),
+        }}
+      />
+      <Input
+        id="applicationUrlFr"
+        name="applicationUrlFr"
+        label={intl.formatMessage(formLabels.applicationUrlFr)}
+        type="text"
+        rules={{
+          required: intl.formatMessage(errorMessages.required),
+        }}
+      />
```

Comment:
Outside of scope, but should we add a URL type to our inputs at some point?

![image](https://github.com/user-attachments/assets/cbc13a30-b74d-4228-abe3-50015c3c7f95)

It appears to be supported though

![image](https://github.com/user-attachments/assets/e2209141-5f6d-4a03-b44a-a2feebb45b25)

Reply:
I started an issue for this.
https://github.com/GCTC-NTGC/gc-digital-talent/issues/12080